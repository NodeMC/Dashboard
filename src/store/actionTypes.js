export const LOGIN = "LOGIN"; // eslint-disable-line
