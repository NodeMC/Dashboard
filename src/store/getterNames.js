export const loginFailed = "loginFailed"; // eslint-disable-line
