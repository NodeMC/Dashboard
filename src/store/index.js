import Vue from "vue";
import Vuex from "vuex";

import auth from "./modules/auth";

Vue.use(Vuex);

const DEBUG = process.env.NODE_ENV === "development";

export default new Vuex.Store({
    modules: {
        auth,
    },
    strict: DEBUG,
});
