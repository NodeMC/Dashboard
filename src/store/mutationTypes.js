export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOGIN_ERROR = "LOGIN_ERROR";
export const LOGIN_START = "LOGIN_START";
export const LOGIN_FINISH = "LOGIN_FINISH";
