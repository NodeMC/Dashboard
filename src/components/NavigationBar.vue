<template>
    <nav class="navbar is-primary" role="navigation">
        <div class="navbar-brand">
            <router-link class="navbar-item is-size-4" to="/">NodeMC</router-link>
            <button class="button navbar-burger is-primary is-radiusless"
                :class="{ 'is-active': hamburgerActive }"
                @click="toggleHamburger">
                <span></span>
                <span></span>
                <span></span>
            </button>
        </div>
        <div class="navbar-menu"
            :class="{ 'is-active': hamburgerActive }">
            <div class="navbar-start">
                <!-- This space left intentionally blank -->
            </div>
            <div class="navbar-end">
                <router-link class="navbar-item"
                    to="/login"
                    v-if="!loggedIn">
                    <b-icon icon="person"></b-icon>
                    <span>
                        Log in
                    </span>
                </router-link>
                <div class="navbar-item has-dropdown is-hoverable"
                    to="/user"
                    v-else>
                    <a class="navbar-link">
                        <b-icon icon="person"></b-icon>
                        <span>Welcome, {{username}}</span>
                    </a>
                    <div class="navbar-dropdown">
                        <a class="navbar-item" @click="logout">
                            <b-icon icon="exit_to_app"></b-icon>
                            <span>Log out</span>
                        </a>
                    </div>
                </div>
            </div>
        </div>
    </nav>
</template>

<script>
import * as actionTypes from "@/store/actionTypes";

export default {
    name: "navigation",
    data() {
        return {
            hamburgerActive: false,
        };
    },
    computed: {
        loggedIn() {
            return this.$store.state.auth.loggedIn;
        },
        username() {
            return this.$store.state.user.username;
        },
    },
    methods: {
        toggleHamburger() {
            this.hamburgerActive = !this.hamburgerActive;
        },
        logout() {
            this.$store.dispatch(actionTypes.LOGOUT);
        },
    },
};
</script>

<style lang="scss" scoped>
button.navbar-burger {
    border: none;
}
</style>

