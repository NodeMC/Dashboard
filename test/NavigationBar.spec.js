import { shallow, createLocalVue } from "vue-test-utils";
import { createRenderer } from "vue-server-renderer";
import Vuex from "vuex";

import NavigationBar from "@/components/NavigationBar";
import * as actionTypes from "@/store/actionTypes";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("NavigationBar.vue when logged in", () => {
    let wrapper;

    let actions;
    let state;
    let store;

    beforeEach(() => {
        actions = {
            [actionTypes.LOGOUT]: jest.fn(),
        };
        state = {
            auth: {
                loggedIn: true,
            },
            user: {
                username: "Username",
            },
        };
        store = new Vuex.Store({
            actions,
            state,
        });

        wrapper = shallow(NavigationBar, { store, localVue });
    });

    it("matches snapshot", () => {
        const renderer = createRenderer();
        renderer.renderToString(wrapper.vm, (err, str) => {
            if (err) throw new Error(err);
            expect(str).toMatchSnapshot();
        });
    });

    it("has a user navbar item", () => {
        const item = wrapper.find(".navbar-menu > .navbar-end > .navbar-item");
        const label = item.find(".navbar-link span");
        expect(label.text()).toBe(`Welcome, ${state.user.username}`);
    });

    it("has a user dropdown", () => {
        const item = wrapper.find(".navbar-menu > .navbar-end > .navbar-item");
        expect(item.hasClass("has-dropdown")).toBe(true);
        expect(item.findAll(".navbar-dropdown")).toHaveLength(1);
        expect(item.findAll(".navbar-dropdown > .navbar-item")).toHaveLength(1);
    });
});
