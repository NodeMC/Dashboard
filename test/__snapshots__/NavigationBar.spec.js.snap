// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar.vue when logged in matches snapshot 1`] = `
<nav role="navigation" class="navbar is-primary">
    <div class="navbar-brand">
        <router-link to="/" class="navbar-item is-size-4">NodeMC</router-link> <button class="button navbar-burger is-primary is-radiusless"><span></span> <span></span> <span></span></button></div>
    <div class="navbar-menu">
        <div class="navbar-start"></div>
        <div class="navbar-end">
            <div to="/user" class="navbar-item has-dropdown is-hoverable"><a class="navbar-link"><b-icon icon="person"></b-icon> <span>Welcome, Username</span></a>
                <div class="navbar-dropdown"><a class="navbar-item"><b-icon icon="exit_to_app"></b-icon> <span>Log out</span></a></div>
            </div>
        </div>
    </div>
</nav>
`;
